yt_dlp/utils.py

```diff
@@ -5414,6 +4812,231 @@ def __repr__(self):
+        if compat_urlparse.urlparse(proxy).scheme.lower() in ('socks', 'socks4', 'socks4a', 'socks5'):
+            req.add_header('Ytdl-socks-proxy', proxy)
+            # yt-dlp's http/https handlers do wrapping the socket with socks
+            return None
+        return compat_urllib_request.ProxyHandler.proxy_open(
+            self, req, proxy, type)
+
+
+# TODO: compat (moved to networking.utils)
+def random_user_agent():
+    from .networking.utils import random_user_agent as r
+    return r()
+
+
+# TODO: compat (moved to networking.utils)
+SUPPORTED_ENCODINGS = []
+
+# TODO: compat (moved to networking.utils)
+std_headers = {}
+
+# TODO: compat (moved to networking.utils)
+USER_AGENTS = {}
+
+
+# TODO: compat (moved to networking._urllib)
+def sanitized_Request(*args, **kwargs):
+    from .networking._urllib import sanitized_Request
+    return sanitized_Request(*args, *kwargs)
+
+
+# TODO: compat (moved to networking._urllib)
+def _create_http_connection(*args, **kwargs):
+    from .networking._urllib import _create_http_connection as c
+    return c(*args, **kwargs)
+
+
+# TODO: compat (moved to networking.utils)
+def handle_youtubedl_headers(headers):
+    from .networking.utils import handle_youtubedl_headers as h
+    return h(headers)
+
+
+# TODO: compat (moved to networking._urllib)
+def YoutubeDLHandler(*args, **kwargs):
+    from .networking._urllib import YoutubeDLHandler as y
+    return y(*args, **kwargs)
+
+
+# TODO: compat (moved to networking._urllib)
+def make_socks_conn_class(base_class, socks_proxy):
+    from .networking._urllib import make_socks_conn_class
+    return make_socks_conn_class(base_class, socks_proxy)
+
+
+# TODO: compat (moved to networking._urllib)
+def YoutubeDLHTTPSHandler(*args, **kwargs):
+    from .networking._urllib import YoutubeDLHTTPSHandler
+    return YoutubeDLHTTPSHandler(*args, *kwargs)
+
+
+# TODO: compat (moved to networking._urllib)
+def YoutubeDLCookieProcessor(*args, **kwargs):
+    from .networking._urllib import YoutubeDLCookieProcessor
+    return YoutubeDLCookieProcessor(*args **kwargs)
+
+
+# TODO: compat (moved to networking._urllib)
+def YoutubeDLRedirectHandler(*args, **kwargs):
+    from .networking._urllib import YoutubeDLRedirectHandler
+    return YoutubeDLRedirectHandler(*args, **kwargs)
+
+
+# TODO: compat (moved to networking._urllib)
+def HEADRequest(*args, **kwargs):
+    from .networking._urllib import HEADRequest
+    return HEADRequest(*args, **kwargs)
+
+
+# TODO: compat (moved to networking._urllib)
+def PUTRequest(*args, **kwargs):
+    from .networking._urllib import PUTRequest
+    return PUTRequest(*args, **kwargs)
+
+
+# TODO: compat (moved to networking._urllib)
+def update_Request(*args, **kwargs):
+    from .networking._urllib import update_Request
+    return update_Request(*args, **kwargs)
+
+
+def YoutubeDLCookieJar(*args, **kwargs):
+    from cookies import YoutubeDLCookieJar
+    return YoutubeDLCookieJar(*args, **kwargs)
+
```

Comment:
@pukkandan how should we deal with all these functions/classes in utils.py that have been moved, or are no longer used?